marathon/src/main/scala/io/buoyant/marathon/v2/Api.scala

```diff
@@ -101,21 +112,90 @@ object Api {
   }
 }
 
-private class AppIdApi(client: Api.Client, apiPrefix: String)
+private class AppIdApi(client: Api.Client, apiPrefix: String, auth: Option[Api.Auth])
   extends Api
   with Closable {
 
   import Api._
 
+  var authToken: Option[String] = None
+
   def close(deadline: Time) = client.close(deadline)
 
-  def getAppIds(): Future[Api.AppIds] = {
-    val req = http.Request(s"$apiPrefix/apps")
-    Trace.letClear(client(req)).flatMap(rspToApps(_))
-  }
+  private[this] def refreshToken(): Future[String] =
```

Comment:
i was torn between having this return a fresh token, or just updating `authToken`, so i'm doing both. thoughts?